### ISSUE TITLE
can't resolve merge conflicts with --install.frozen-lockfile in .yarnrc

### ISSUE THREAD

**Issue by mxmul** [yarnpkg/yarn#5382]
**Do you want to request a *feature* or report a *bug*?**
feature?

**What is the current behavior?**

If a project has a .yarnrc file that specifies `--install.frozen-lockfile false`, then you can't take advantage of the automatic merge conflict resolution on `yarn install`.

**What is the expected behavior?**

Yarn either provides a way to override the settings in .yarnrc during `yarn install` (e.g. `yarn install --no-frozen-lockfile`), or provides a new CLI command to explicitly resolve merge conflict markers (e.g. `yarn resolve-merge`).

**Comment by ispringer** [comment]
@mxmul, does deleting yarn.lock prior to running yarn install work?

**Comment by mxmul** [comment]
@ispringer Yeah that does resolve the conflict, but is a pretty heavy hammer – I'd prefer to update only the dependencies affected by my branch(es).

My work-around has been something like `rm .yarnrc && yarn && git checkout .yarnrc`, but that's clearly not ideal either.

**Comment by BYK** [comment]
Thanks for bringing this up. I think we need a way to turn this off too. Would you like to give a try at implementing this?

**Comment by mxmul** [comment]
@BYK I could work on this, but I curious if you have any opinions on how this should be implemented. I think my preference would be a new command to explicitly resolve the conflicts (`yarn resolve-merge`) that doesn't use the `--install.frozen-lockfile` configuration.

**Comment by arcanis** [comment]
I think implementing a `--mutable-lockfile` option should be enough, what do you think?

**Comment by mxmul** [comment]
That would work as well, but the behavior is a bit less obvious to me (do command-line options always take precedence over .yarnc options? what would something like `yarn --frozen-lockfile --mutable-lockfile` do?).

**Comment by arcanis** [comment]
Command line options overriding each other has precedent, for example with C compilers. That being said, we currently use Commander, which is really inefficient, so it won't be possible to have the exact same behavior where the last one wins (one of the options will have to take precedence over the others).

**Comment by alexeyten** [comment]
I would suggest `--no-frozen-lockfile` or `--frozen-lockfile=false`.

I have similar problem with `--ignore-optional` and would like to have uniform solution, not to remember mutable-lockfile, regard-optional etc.

**Comment by arcanis** [comment]
Problem is that there isn't only `frozen-lockfile`. We also have `pure-lockfile`. Ideally the best would be something like this; `-c lockfile.state=mutable|pure|frozen`, but I'm not sure this can be easily done with Commander.

That being said, we hope to overhaul our command line for the 2.0, so if someone is interested in migrating to a more modern framework like Yargs, it would be super helpful. We would then just have to alias `--XXX-lockfile` to `-c lockfile.state=XXX` for backward compatibility, deprecate them, and remove them in the next major update.

**Comment by amakhrov** [comment]
For me, the only use case to override the `frozen-lockfile` option is when resolving merge conflicts.
So indeed, introducing as separate command e.g. `yarn resolve-merge` sounds like a clean solution to that.

**Comment by dacioromero** [comment]
I'm on the `--frozen-lockfile=false` boat to be consistent with `--production`. Similar flags (e.g. `--check-files`, `--flat`, etc.) should probably accept booleans too
